server/routes/auth.js

```diff
@@ -0,0 +1,47 @@
+import jwt from 'jsonwebtoken';
+import jwtDecode from 'jwt-decode';
+import model from '../models';
+
+const secret = process.env.SECRET_TOKEN || 'secret';
+
+const verifyToken = (req, res, next) => {
+  const token = req.headers.authorization || req.headers['x-access-token'];
+  if (!token) {
+    return res.status(401)
+      .send({ message: 'Not Authorized' });
+  }
+
+  jwt.verify(token, secret, (error, decoded) => {
+    if (error) {
+      return res.status(401)
+        .send({ message: 'Token Invalid' });
+    }
+
+    req.decoded = decoded;
+    next();
+  });
+};
+
+const decodeToken = (req) => {
+  const token = req.headers.authorization || req.headers['x-access-token'];
+  const decodedToken = jwtDecode(token);
```

Comment:
Doesn't `jwt-decode` do the same thing that `jwt.verify()` does. If it does then why bring on an extra overhead of adding another module. Is the statement about validation in the image below not scary?
![selection_024](https://cloud.githubusercontent.com/assets/17332992/23168278/ee5afb38-f847-11e6-9046-77de07398c60.png)